### PR TITLE
added interface for string

### PR DESCRIPTION
**Pull request by selwiak**
[added interface for string](https://github.com/selwiak/GL-cpp-training/commit/f4cf3d3756c4f2657b2afcb82cdbac346bea60a1)

### CODE REVIEW

File: mystring.h [hunk 1]
```diff
@@ -1,7 +1,19 @@
 #include <iostream>
 #include <memory>
 
-class String {
+class StringInterface {
```
Comment (dmytro-dubovyk) [inline mystring.h]: I suggest putting the interface into a separate file

Comment (dmytro-dubovyk) [inline mystring.h]: also it's more typical to name interfaces in some clear (but not overwhelming) pattern, here I suggest simply "IString"

File: main.cpp [hunk 2]
```diff
@@ -8,26 +8,23 @@ int main()
     char temp[] = "Hello";
     char temp2[] = "world";
 
-
     String s1{temp};
```
Comment (dmytro-dubovyk) [inline main.cpp]: now use here something like std::unique_ptr<IString> s1{new String(temp)}

File: mystring.h [hunk 3]
```diff
@@ -1,7 +1,19 @@
 #include <iostream>
 #include <memory>
 
-class String {
+class StringInterface {  
+    public:
+    virtual ~StringInterface() = 0;
```
Comment (dmytro-dubovyk) [inline mystring.h]: destructors better not make abstract, use = default instead

**Review by dmytro-dubovyk — CHANGES_REQUESTED** [review]

File: mystring.h (hunk 1 above)
Comment (selwiak) [reply mystring.h]: done

Comment (selwiak) [reply mystring.h]: done

File: mystring.h (hunk 3 above)
Comment (selwiak) [reply mystring.h]: done

File: main.cpp (hunk 2 above)
Comment (selwiak) [reply main.cpp]: done

File: main.cpp [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #include <iostream>
+#include <memory>
```
Comment (dmytro-dubovyk) [inline main.cpp:2]: this is actually not needed as you already have it included in IString.h

**Pull request merged**